src/js/components/uploadDetachedFiles/uploadDetachedFileMeta.jsx

```diff
@@ -0,0 +1,458 @@
+			});
+		} else {
+			this.setState({
+				showDatePicker: false,
+				showUploadFilesBox: false
+			});
+		}
+	}
+
+	handleDateChange(file, date, dateType) {
+		// merge the new date into the file's state without affecting the other keys
+		const newState = Object.assign(this.state[file], {
+			[dateType]: moment(date)
+		});
+
+		this.setState({
+			[file]: newState
+		}, () => {
+			this.validateDates(file);
+		});
+	}
+
+	validateDates(file) {
+		// validate that dates are provided for both fields and the end dates don't come before the start dates
+		let state = "incomplete";
+
+		const dFile = Object.assign({}, this.state[file]);
+
+		// validate the date ranges
+		const start = this.state[file].startDate;
+		const end = this.state[file].endDate;
+		if (start && end) {
+			// both sets of dates exist
+			if (!end.isSameOrAfter(start)) {
+				// end date comes before start date, invalid
+				// show an error message
+				dFile.error = {
+					show: true,
+					header: 'Invalid Dates',
+					description: 'The end date cannot be earlier than the start date.'
+				};
+				dFile.valid = false;
+			}
+			else {
+				// valid!
+				dFile.error = {
+					show: false,
+					header: '',
+					description: ''
+				};
+				dFile.valid = true;
+			}
+		}
+		else {
+			// not all dates exist yet
+			dFile.error = {
+				show: false,
+				header: '',
+				description: ''
+			};
+			dFile.valid = false;
+		}
+
+		this.setState({
+			[file]:dFile,
+			showUploadFilesBox: dFile.valid
+		});
+	}
+
+	showError(file, header, description) {
+		// Show any error that occurs at any point during file upload
+		const state = Object.assign({}, this.state[file], {
+			error: {
+				show: true,
+				header: header,
+				description: description
+			}
+		});
+		
+		this.setState({
+			[file]: state
+		});
+	}
+
+	hideError(file) {
+		// Stop displaying the error for the given file
+		const state = Object.assign({}, this.state[file], {
+			error: {
+				show: false,
+				header: '',
+				description: ''
+			}
+		});
+
+		this.setState({
+			[file]: state
+		});
+	}
+
+	uploadFile() {
+		// upload specified file
+		this.props.setSubmissionState('uploading');
+		this.props.submission.meta['startDate'] = this.state.detachedAward.startDate.format('DD/MM/YYYY');
+		this.props.submission.meta['endDate'] = this.state.detachedAward.endDate.format('DD/MM/YYYY');
+		this.props.submission.meta['subTierAgency'] = this.state.agency;
+		
+		if (kGlobalConstants.LOCAL == true && !this.isUnmounted) {
+			UploadHelper.performDetachedLocalUpload(this.props.submission)
+                .then((submissionID) => {
+                    // TODO: Remove this when this is eventually tied to user accounts
+                    this.props.setSubmissionId(submissionID);
+					this.checkFileStatus(submissionID);
+					this.props.history.push('/uploaddetachedfiles/'+submissionID);
+					this.props.validate();
+                })
+				.catch((err) => {
+					if (err.httpStatus == 403) {
```

Comment:
You can probably turn this into one call and dynamically set `notAllowed` based on `err.httpStatus`